examples/ota-provider-app/linux/main.cpp

```diff
@@ -209,6 +221,32 @@ bool HandleOptions(const char * aProgram, OptionSet * aOptions, int aIdentifier,
             retval = false;
         }
         break;
+    case kOptionDiscriminator: {
+        uint16_t discriminator = static_cast<uint16_t>(strtol(aValue, NULL, 0));
```

Comment:
This should be `strtoul` (and same for all the other uint16_t/uint32_t

Reply:
Using `strtoul` for converting string to unsigned integers.